components/canvas/canvas_data.rs

```diff
@@ -20,8 +20,13 @@ use webrender_api;
 
 pub struct CanvasData<'a> {
     drawtarget: DrawTarget,
-    /// TODO(pcwalton): Support multiple paths.
-    path_builder: PathBuilder,
+    /// User-space path builder.
+    path_builder: Option<PathBuilder>,
+    /// Device-space path builder, if transforms are added during path building.
+    device_space_path_builder: Option<PathBuilder>,
```

Comment:
Why do we need to store the user-space path? Could we just store the path in device space?